docs/DeveloperGuide.md

```diff
@@ -626,7 +626,161 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 --------------------------------------------------------------------------------------------------------------------
 
-## **Appendix: Instructions for manual testing**
+## **Appendix B: Planned Enhancements**
+
+### Customisable fields
+
+**Potential Issues With Current Feature**
+
+Existing `Person` class has all fields: `name`, `age`, `sex`, `ic`, `phone`, `email` and `address`. However, users might not require all of these fields. 
+User A might not require the `phone` field while User B might require all fields, including those that ClinicMate has yet to implement. 
+Hence, it is important that ClinicMate allows for users to customize the fields they require to make it less restrictive and allow for flexibility.
+
+**Proposed Enhancements**
+
+Before users initialize their own version of ClinicMate, they will be able to choose the fields from 
+a whole range of fields which we offer. The fields could also be made optional depending on user's choice. 
+Their application will thus be initialized with user's very own fields which they require.
+
+**Examples**
+
+- User A could have the `Person` class with only `name`, `age`, `sex`, and `ic`.
+- User B could have the `Person` class with only `name`, `age`, `sex`,` ic` and `phone` with the `phone` field optional.
+
+
+### Flexible find command 
+
+**Potential Issues With Current Feature**
+
+Currently, the `find` command only allows users to find patient details using their `IC_NUMBER`. This could be inconvenient and troublesome 
+at times as users are required to type out the full `IC_NUMBER`. A single wrong character thus requires the user to re-look at the whole string of
+`IC_NUMBER` again in order to identify their mistakes.
+
+**Proposed Enhancements**
+
+In light of this issue, we would be working on our application to provide an update to the `find` command 
+to allow users to find contacts using different fields such as `name`, `age`, `email` etc. They will be case-insensitive to allow for better search results.
+
+**Examples**
+
+- find `John` returns contact with `name` of `John Doe` and corresponding `IC_NUMBER` of `T0123456A`
+- find `johndoe@mail.com` returns contact with `email` of `johndoe@mail.com` and corresponding `IC_NUMBER` of `T0123456A`
+
+### Update notes window during execution of `find` command
```

Comment:
I think the update notes window is not referring to the `find` command because the `find` command does change the contents of out patient notes panel (the right hand side pane). 

This is more of after using find T0123456A then maybe u do addnote to S1234567A--> the patient notes panel will not show the updated note for S1234567A but will still show the note for T0123456A.

Then we wna say that we will want to fix this such that if we interact with any of the details of any patients (thru addnote/etc) then the most updated note will be displayed